src/Cursor.re

```diff
@@ -1,34 +1,46 @@
-type t = {
-  line: int,
-  column: int,
-};
+open EditorCoreTypes;
+
+type t =
+  Location.t = {
+    line: Index.t,
+    column: Index.t,
+  };
```

Comment:
I've aliased `Cursor.t` to `Location.t` here since they're effectively the same type. But I'm not sure `Cursor.t` is even needed  since most of the functions here either operate on `Location.t` directly or its components.

Reply:
This sounds reasonable!

>  But I'm not sure Cursor.t is even needed since most of the functions here either operate on Location.t directly or its components.

Makes sense at this time. There will be a couple extra concepts that we add to the Cursor later, so support more scenarios (mainly multi-cursor)
- `curswant` - this is a Vim setting in addition to the position. It marks the column the cursor 'wants' to move to when moving between lines of different length. If you're at the end of the line, for example, it sets it to a constant so that it knows to move to the end of other lines as you navigate up and down. However, if you've moved down from the 2nd column, you likely want to stay at the second column - this behavior is recorded in `curswant`.
- `selection` (range) - for multiple cursors in visual / select / replace modes, we'll need a concept of _per-cursor_ selection ranges.

So leaving the `Cursor.t` type sounds good, and we can extend it later to support these additional features w/ multi-cursor